homeassistant/components/contec_controllers/config_flow.py

```diff
@@ -0,0 +1,96 @@
+"""Config flow for Contec Controllers integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+from typing import Any
+
+from ContecControllers.ContecConectivityConfiguration import (
+    ContecConectivityConfiguration,
+)
+from ContecControllers.ControllerManager import ControllerManager
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+from .contec_tracer import ContecTracer
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("ip"): str,
+        vol.Required("numberOfControllers"): int,
+        vol.Required("port"): int,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> dict[str, Any]:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+
+    numberOfControllers: int = data["numberOfControllers"]
```

Comment:
Is there a way to discover the number of controllers without asking the user?

Reply:
unfortunately, no... it doesn't exposed in any way...